frontend/src/metabase/static-viz/components/XYChart/Values/Values.tsx

```diff
@@ -118,25 +124,41 @@ export default function Values({
             return null;
           }
 
-          const { xAccessor, yAccessor } = getXyAccessors(
+          const { xAccessor, yAccessor, dataYAccessor } = getXyAccessors(
             value.series.type,
             value.xScale,
             value.yScale,
             getBarXOffset(value.series),
             value.flipped,
           );
 
+          const shouldRenderDataPoint = (
+            ["line", "area"] as VisualizationType[]
+          ).includes(value.series.type);
           return (
-            <OutlinedText
-              key={index}
-              x={xAccessor(value.datum)}
-              y={yAccessor(value.datum)}
-              textAnchor="middle"
-              verticalAnchor="end"
-              {...valueProps}
-            >
-              {formatter(getY(value.datum), compact)}
-            </OutlinedText>
+            <>
+              <OutlinedText
+                key={index}
+                x={xAccessor(value.datum)}
+                y={yAccessor(value.datum)}
+                textAnchor="middle"
+                verticalAnchor="end"
+                {...valueProps}
+              >
+                {formatter(getY(value.datum), compact)}
+              </OutlinedText>
+              {shouldRenderDataPoint && (
```

Comment:
only render the data point circle on visible data point values, not sure if this is going to be a problem, but I revert my decision to display the data point circles on every data point since if there's too many data points then the circle is going to overlap and I haven't figured out how app viz determine when to not display the circle yet.
![image](https://user-images.githubusercontent.com/1937582/189135485-7891dd53-9705-44eb-a105-975e56046fc2.png)